CMake/SetupFfmpeg.cmake

```diff
@@ -41,6 +41,8 @@ if(MACOSX)
     list(APPEND FFMPEG_CONFIGURE "--arch=arm64" "--extra-cflags=-arch arm64" "--extra-ldflags=-arch arm64")
   elseif(CMAKE_OSX_ARCHITECTURES STREQUAL "x86_64")
     list(APPEND FFMPEG_CONFIGURE "--arch=x86_64" "--extra-cflags=-arch x86_64" "--extra-ldflags=-arch x86_64")
+  elseif(CMAKE_OSX_ARCHITECTURES STREQUAL "arm64;x86_64" OR CMAKE_OSX_ARCHITECTURES STREQUAL "x86_64;arm64")
+    list(APPEND FFMPEG_CONFIGURE "--arch=arm64,x86_64" "--extra-cflags=-arch arm64 -arch x86_64" "--extra-ldflags=-arch arm64 -arch x86_64")
```

Comment:
this won't work, please check how `--arch` option for [ffmpeg's `configure` script works](https://github.com/FFmpeg/FFmpeg/blob/master/configure). I'm still waiting for the pipeline to fail, but it will be here: https://github.com/florczakraf/itgmania/actions/runs/10356546702/job/28666613148

Reply:
Ugh, this file was giving me a lot of trouble on FreeBSD as well. 

This is the suggested way to set up a universal build when dealing with `CMAKE_OSX_ARCHITECTURES`, but it looks like ffmpeg will need special handling, so if you're able to find a way to get ffmpeg to cooperate before I do then that would be excellent, otherwise I'll set this to a draft until that's sorted out.

Reply:
I don't have a mac, so debugging this over github pipelines is definitely something I'm going to waste time on

Reply:
> I don't have a mac, so debugging this over github pipelines is definitely something I'm not going to waste time on

Okay well thanks anyway

Reply:
> This is the suggested way to set up a universal build when dealing with `CMAKE_OSX_ARCHITECTURES`

I suppose it works well for standard cmake-managed projects. However, ffmpeg doesn't use cmake for building, we only have a thin wrapper for their build system glued. We run their `configure` script during our `build` step so it fails really late in the process